server/db_mysql.go

```diff
@@ -129,6 +131,122 @@ func (ms *MsqlCache) NextFixity(cutoff time.Time) string {
 	return item
 }
 
+// GetFixityById
+func (ms *MsqlCache) GetFixityById(id string)  *fixity {
+        const query = `
+                SELECT  id, item, scheduled_time, status, notes
+                FROM fixity
+                WHERE id = ?
+                LIMIT 1`
+
+	var thisFixity = new(fixity)
+	var thisWhen mysql.NullTime
+
+        err := ms.db.QueryRow(query, id).Scan(&thisFixity.Id, &thisFixity.Item, &thisWhen, &thisFixity.Status, &thisFixity.Notes)
+        if err == sql.ErrNoRows {
+                // no next record
+                return nil
+        } else if err != nil {
+                log.Println("GetFixtyByID  MySQL queryrow", err.Error())
+                return nil
+        }
+
+        // Handle for nil time value
+	if thisWhen.Valid {
+	    thisFixity.Scheduled_time = thisWhen.Time
+	} else {
+	    thisFixity.Scheduled_time = time.Time{}
+	}
+
+        log.Println("id= ", thisFixity.Id, "scheduled_time= ", thisFixity.Scheduled_time, "status= ", thisFixity.Status, "notes= ", thisFixity.Notes)
+        return thisFixity
+}
+
+// GetFixity
+func (ms *MsqlCache) GetFixity( start string, end string, item string, status string)  []*fixity {
+        query := buildQuery(start, end, item, status)
+	log.Println("GET /fixity query= ", query)
+	var thisWhen mysql.NullTime
+        var fixityResults []*fixity
+
+        rows, err := ms.db.Query(query)
+        if err == sql.ErrNoRows {
+                // no next record
+                return nil
+        } else if err != nil {
+                log.Println("GetFixity Query MySQL", err.Error())
+                return nil
+        }
+        defer rows.Close()
+
+	for rows.Next() {
+		var thisFixity = new(fixity)
+		scanErr := rows.Scan(&thisFixity.Id, &thisFixity.Item, &thisWhen, &thisFixity.Status, &thisFixity.Notes)
+		if scanErr != nil {
+		    log.Println("GetFixity Scan MySQL", err.Error())
+                    return nil
+		}
+        	// Handle for nil time value
+		if thisWhen.Valid {
+	    		thisFixity.Scheduled_time = thisWhen.Time
+		} else {
+	    		thisFixity.Scheduled_time = time.Time{}
+		}
+                log.Println("id= ", thisFixity.Id, "item =", thisFixity.Item, "scheduled_time", thisFixity.Scheduled_time, "status= ", thisFixity.Status, "notes= ", thisFixity.Notes)
+		fixityResults = append(fixityResults,thisFixity)
+        }
+
+        return fixityResults
+}
+
+// construct an return an sql query, using the parameters passed
+func buildQuery( start string, end string, item string, status string) string {
+	var query bytes.Buffer
+	query.WriteString("SELECT  id, item, scheduled_time, status, notes FROM fixity")
+
+	params := []string{ "start", "end", "item", "status"}
+
+	conjunction := " WHERE "
+
+	for _, param := range params {
```

Comment:
This is a curious control construct. It iterates through a constant string array, and then dispatchs on each possible value. It seems like everything could be inlined. The update of the conjunction variable could be done in each of the `if` clauses.